calcom/cal.com
# feat: Cal AI V1.1.0

[LOOM VIDEO](https://www.loom.com/share/1402b26c101e40b88bbd2631fd4fdf40?sid=6355551e-ad77-4942-acd6-2a020e520722)

## What does this PR do?



This PR adds features to Cal AI:

- [x] Multi user support (book meetings with 2 user availabilities and invite cal users by @username)

* In this example, Sarim is a cal user in Phoenix that only works on Wednesdays.
<img width="684" alt="Screenshot 2023-09-22 at 3 12 47 PM" src="https://github.com/calcom/cal.com/assets/36115192/d74cf072-11a9-41ef-9637-b510d772943d">

- [x] Booking Links (send booking links via email to non cal.com users)

* In this example, Sarim is not a cal user so a booking link is emailed to them.
<img width="685" alt="Screenshot 2023-09-22 at 3 04 20 PM" src="https://github.com/calcom/cal.com/assets/36115192/6b62e4be-6d96-4b99-9c3f-a47b79794724">

* A booking link is sent to Sarim with the primary user CC'd.
<img width="685" alt="Screenshot 2023-09-22 at 3 04 34 PM" src="https://github.com/calcom/cal.com/assets/36115192/9357efe1-abb7-4b71-b5b3-8152710e0218">

- [x] Well formatted emails (not quite markdown, but spacing, signature and links are supported)



[Loom Video](https://www.loom.com/share/1402b26c101e40b88bbd2631fd4fdf40?sid=e14d26b2-717a-449b-bb5f-01def9b611a5)

## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?



New env variables:
FRONTEND_URL=http://localhost:3000 (dev) or https://cal.com (prod)

Run the backend, frontend and ai app following [the docs](https://github.com/calcom/cal.com/blob/main/apps/ai/README.md).
Install the cal app for your user email
Send an email to <ANYTHING>@cal.ai (or your dev SendGrid inbound parse email)

Try 
- "Book a meeting with @usa next week at a good time"
- "Send a booking link to hi@example.com"

Expect
- A meeting booked at a time when both USA and you are free (on your calendar with one of your event types)
- An email from cal ai sent to hi@example.com with a booking link (you are cc'd)

## Mandatory Tasks


## Checklist



- ~I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)~
- ~My code doesn't follow the style guidelines of this project~
- ~I haven't commented my code, particularly in hard-to-understand areas~
- ~I haven't checked if my PR needs changes to the documentation~
- ~I haven't checked if my changes generate no new warnings~
- ~I haven't added tests that prove my fix is effective or that my feature works~
- ~I haven't checked if new and existing unit tests pass locally with my changes~
